streams/src/main/java/org/apache/kafka/streams/state/BuiltInDslStoreSuppliers.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.state;
+
+import org.apache.kafka.streams.kstream.EmitStrategy;
+import org.apache.kafka.streams.state.internals.RocksDbIndexedTimeOrderedWindowBytesStoreSupplier;
+import org.apache.kafka.streams.state.internals.RocksDbTimeOrderedSessionBytesStoreSupplier;
+
+public class BuiltInDslStoreSuppliers {
```

Comment:
I think we should probably have javadocs for this class and the two built-in implementations. They can be brief but should cover at least how/where to use them:
1. how to configure an app-wide global default (including the config name)
2. how to configure individual operators via Materialized and StreamJoined
3. the hierarchy of preference if specified in multiple places (ie StreamJoined/Materialized overrides configured global default overrides unconfigured Streams default

Reply:
Actually maybe the stuff I listed above should be in the `DslStoreSuppliers` interface javadocs instead of here. But I still think this class should have javadocs, even just a sentence or two